catalog/app/utils/PackageUri.spec.js

```diff
@@ -0,0 +1,179 @@
+import * as PackageUri from './PackageUri'
+
+describe('utils/PackageUri', () => {
+  describe('parse', () => {
+    it('should work for a valid URI w/o tag, hash or path', () => {
+      expect(PackageUri.parse('quilt+s3://bucket-name#package=quilt/test')).toEqual({
+        bucket: 'bucket-name',
+        name: 'quilt/test',
+      })
+    })
+
+    it('should work for a valid URI w/ tag', () => {
+      expect(
+        PackageUri.parse('quilt+s3://bucket-name#package=quilt/test:latest'),
+      ).toEqual({
+        bucket: 'bucket-name',
+        name: 'quilt/test',
+        tag: 'latest',
+      })
+    })
+
+    it('should work for a valid URI w/ hash and path', () => {
+      expect(
+        PackageUri.parse(
+          'quilt+s3://bucket-name#package=quilt/test@abc1&path=sub%2Fpath',
+        ),
+      ).toEqual({
+        bucket: 'bucket-name',
+        name: 'quilt/test',
+        hash: 'abc1',
```

Comment:
Actually, it is unexpected that `#part` is not hash. It conflicts with https://nodejs.org/dist/latest-v14.x/docs/api/url.html#url_url_hash

I don't know how to resolve it. I think we should consider replacing it with a different word on the project level.

Reply:
URI "hash" and package "hash" are two completely different things, and the format of the URI has been discussed and decided upon in the relevant spec

Reply:
I understand. But they use the same word and this is confusing. Different things - one word